http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpWsmDao.scala

```diff
@@ -201,28 +211,77 @@ class HttpWsmDao[F[_]](httpClient: Client[F], config: HttpWsmDaoConfig)(implicit
+   * matches a given resource type and purpose. Throws an error if the resource is not found.
+   *
+   * @param landingZoneResourcesByPurpose the collection in which to search
+   * @param resourceType                  type of resource to return
+   * @param purpose                       purpose of resource to return
+   * @return the LandingZoneResource
+   */
+  private def getLandingZoneResource(
     landingZoneResourcesByPurpose: Map[(LandingZoneResourcePurpose, String), List[LandingZoneResource]],
     resourceType: String,
-    purpose: LandingZoneResourcePurpose,
-    useParent: Boolean
-  ): F[String] =
+    purpose: LandingZoneResourcePurpose
+  ): F[LandingZoneResource] =
     landingZoneResourcesByPurpose
       .get((purpose, resourceType.toLowerCase))
       .flatMap(_.headOption)
-      .flatMap { r =>
-        if (useParent) r.resourceParentId.flatMap(_.split('/').lastOption)
-        else r.resourceName.orElse(r.resourceId.flatMap(_.split('/').lastOption))
-      }
       .fold(
-        F.raiseError[String](
+        F.raiseError[LandingZoneResource](
           AppCreationException(s"${resourceType} resource with purpose ${purpose} not found in landing zone")
         )
       )(F.pure)
 
+  /**
+   * Given a collection of landing zone resources by purpose, return the name of the single resource that
+   * matches a given resource type and purpose. Throws an error if the resource is not found.
+   *
+   * @param landingZoneResourcesByPurpose the collection in which to search
+   * @param resourceType                  type of resource to return
+   * @param purpose                       purpose of resource to return
+   * @param useParent                     whether to return the resource's name or its parent's name
+   * @return name of the specified resource
+   */
+  private def getLandingZoneResourceName(
+    landingZoneResourcesByPurpose: Map[(LandingZoneResourcePurpose, String), List[LandingZoneResource]],
+    resourceType: String,
+    purpose: LandingZoneResourcePurpose,
+    useParent: Boolean
+  ): F[String] =
+    OptionT(
+      getLandingZoneResource(landingZoneResourcesByPurpose, resourceType, purpose)
+        .map(res => getLandingZoneResourceName(res, useParent))
+    )
+      .getOrRaise(
+        AppCreationException(s"${resourceType} resource with purpose ${purpose} not found in landing zone")
+      )
+
+  /**
+   * Given a landing zone resource, return that resource's name
+   *
+   * @param resource  the resource whose name to return
+   * @param useParent whether to return the resource's name or its parent's name
+   * @return name of the resource, or None if the name could not be determined
+   */
+  private def getLandingZoneResourceName(resource: LandingZoneResource, useParent: Boolean): Option[String] =
+    if (useParent) resource.resourceParentId.flatMap(_.split('/').lastOption)
+    else resource.resourceName.orElse(resource.resourceId.flatMap(_.split('/').lastOption))
+
```

Comment:
this is a rather verbose refactoring of the original `getLandingZoneResourceName`, to make its constituent parts available to call - for instance, I now need to get a whole LandingZoneResource, not just its name.